modules/imgproc/src/opencl/gaussianBlur3x3.cl

```diff
@@ -0,0 +1,133 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#define DIG(a) a,
+__constant float kx[] = { KERNEL_MATRIX_X };
+__constant float ky[] = { KERNEL_MATRIX_Y };
+
+#define OP(delta, y, x) (convert_float16(arr[(y + delta) * 3 + x]) * ky[y] * kx[x])
+
+__kernel void gaussianBlur3x3_8UC1_cols16_rows2(__global const uint* src, int src_step,
+                                                __global uint* dst, int dst_step, int rows, int cols)
+{
+    int block_x = get_global_id(0);
+    int y = get_global_id(1) * 2;
+    int ssx, dsx;
+
+    if ((block_x * 16) >= cols || y >= rows) return;
```

Comment:
in some cases the global work size can be changed larger in function "bool Kernel::run()" which is in "./modules/core/src/ocl.cpp",  by below code :
    globalsize[i] = ((_globalsize[i] + val - 1)/val)*val;
So it isn't always the value we set in "ocl_GaussianBlur3x3_8UC1()",  the ocl kernel has to check whether block_x * 16 is beyond the actual cols ,  same as the check for y and actual rows.